### PR TITLE
only wrap source if fopen was successful

### LINKED ISSUE #19572: Encryption wrapper - properly handle fopen failure
fopen can return a resource or false - also false should be properly handled: https://github.com/owncloud/core/blob/master/lib/private/files/storage/wrapper/encryption.php#L199-L200

```
{"reqId":"aaa","remoteAddr":"...","app":"PHP","message":"fopen(\/data\/username\/thumbnails\/187309\/36-36.png): failed to open stream: No such file or directory at owncloud\/lib\/private\/files\/storage\/local.php#257","level":3,"time":"2015-09-21T20:24:46-06:00"}
{"reqId":"aaa","remoteAddr":"...","app":"PHP","message":"fwrite() expects parameter 1 to be resource, boolean given at owncloud\/lib\/private\/files\/storage\/wrapper\/encryption.php#200","level":3,"time":"2015-09-21T20:24:46-06:00"}
{"reqId":"aaa","remoteAddr":"...","app":"PHP","message":"fclose() expects parameter 1 to be resource, boolean given at owncloud\/lib\/private\/files\/storage\/wrapper\/encryption.php#201","level":3,"time":"2015-09-21T20:24:46-06:00"}
```

cc @icewind1991 @schiesbn

### PR DESCRIPTION
**Pull request by schiessle**
fix https://github.com/owncloud/core/issues/19572

cc @DeepDiver1975 @MorrisJobke

### CODE REVIEW

**Comment by schiesbn** [comment]
It is just about preventing two follow up error messages, it is not that everything will break it just keeps the log cleaner. In my view nothing urgent but a backport would be also low risk.

Before:

````
Oct 9 12:23:36 rvavdapp23 ownCloud[11187]: {PHP} fopen(/mnt/oc_data/77A57212-EF9F-4E38-AA0D-2E16887650F8/avatar.jpg): failed to open stream: No such file or directory at /var/www/html/owncloud-enterprise/lib/private/files/storage/local.php#257
Oct 9 12:23:36 rvavdapp23 ownCloud[11187]: {PHP} fwrite() expects parameter 1 to be resource, boolean given at /var/www/html/owncloud-enterprise/lib/private/files/storage/wrapper/encryption.php#200
Oct 9 12:23:36 rvavdapp23 ownCloud[11187]: {PHP} fclose() expects parameter 1 to be resource, boolean given at /var/www/html/owncloud-enterprise/lib/private/files/storage/wrapper/encryption.php#201
````

After:

````
Oct 9 12:23:36 rvavdapp23 ownCloud[11187]: {PHP} fopen(/mnt/oc_data/77A57212-EF9F-4E38-AA0D-2E16887650F8/avatar.jpg): failed to open stream: No such file or directory at /var/www/html/owncloud-enterprise/lib/private/files/storage/local.php#257
````

@DeepDiver1975 @karlitschek your decision regarding backport

File: lib/private/files/storage/wrapper/encryption.php [hunk 1]
```diff
@@ -404,6 +408,9 @@ public function fopen($path, $mode) {
 			if ($shouldEncrypt === true && $encryptionModule !== null) {
 				$headerSize = $this->getHeaderSize($path);
 				$source = $this->storage->fopen($path, $mode);
+				if (!is_resource($source)) {
+					return false;
```
Comment (MorrisJobke) [inline lib/private/files/storage/wrapper/encryption.php]: Can the PHPDoc be adjusted then?

**Comment by icewind1991** [comment]
:+1: looks good

File: lib/private/files/storage/wrapper/encryption.php (hunk 1 above)
Comment (schiesbn) [inline lib/private/files/storage/wrapper/encryption.php]: done

**Comment by PVince81** [comment]
:+1: checked that file_put_contents still work (text editor)

**Pull request merged**